### PR TITLE
remove usage of template_file

### PR DESCRIPTION
**Pull request by knrt10**
template_file is deprecated

closes: #196
Signed-off-by: knrt10 <kautilya@kinvolk.io>

### CODE REVIEW

File: assets/terraform-modules/packet/flatcar-linux/kubernetes/controllers.tf [hunk 1]
```diff
@@ -75,13 +75,11 @@ data "ct_config" "controller-ignitions" {
   snippets = var.controller_clc_snippets
 }
 
-data "template_file" "etcds" {
+data "ct_config" "etcds" {
```
Comment (invidian) [inline assets/terraform-modules/packet/flatcar-linux/kubernetes/controllers.tf:78]: What, why do we use `ct_config` for it? Let's break down what we need to generate and figure it out only with Terraform syntax, that should be possible.

**Review by invidian — CHANGES_REQUESTED** [review]

File: assets/terraform-modules/packet/flatcar-linux/kubernetes/controllers.tf (hunk 1 above)
Comment (knrt10) [reply assets/terraform-modules/packet/flatcar-linux/kubernetes/controllers.tf:78]: Yes, I figured it out after seeing the error. Fixing it

File: assets/terraform-modules/bootkube/assets.tf [hunk 2]
```diff
@@ -110,55 +110,50 @@ locals {
 }
 
 # Render kubelet.yaml for kubelet chart
-data "template_file" "kubelet" {
+data "ct_config" "kubelet" {
```
Comment (invidian) [inline assets/terraform-modules/bootkube/assets.tf:113]: `ct_config` stand for "config transpiler" and is related to Flatcar configuration. It has nothing to do with rendering. We should render to `locals` block instead.

**Review by invidian — CHANGES_REQUESTED** [review]

File: assets/terraform-modules/kvm-libvirt/flatcar-linux/kubernetes/bootkube.tf [hunk 3]
```diff
@@ -3,10 +3,10 @@ module "bootkube" {
   cluster_name = var.cluster_name
 
   # Cannot use cyclic dependencies on controllers or their DNS records
-  api_servers          = [data.template_file.controllernames[0].rendered]
+  api_servers          = [for i in range(0, 1) : format("%s-controller-%d.%s", var.cluster_name, i, var.dns_zone)]
```
[the remark this answers is not in the document]
Comment (knrt10) [reply assets/terraform-modules/kvm-libvirt/flatcar-linux/kubernetes/bootkube.tf:6]: It is only rendering the first output whatever the counter_count value is.

[the remark this answers is not in the document]
Comment (knrt10) [reply assets/terraform-modules/kvm-libvirt/flatcar-linux/kubernetes/bootkube.tf:6]:
```
data "template_file" "controllernames" {
  count    = var.controller_count
  template = "$${cluster_name}-controller-$${index}.$${machine_domain}"

  vars = {
    index          = count.index
    cluster_name   = var.cluster_name
    machine_domain = var.dns_zone
  }
}

output "test1" {
  value = [data.template_file.controllernames[0].rendered]
}

output "test2" {
  value = [for i in range(0, 1) : format("%s-controller-%d.%s", var.cluster_name, i, var.dns_zone)]
}

```

To test you can try this

File: assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/profiles.tf [hunk 4]
```diff
@@ -76,15 +74,13 @@ resource "matchbox_profile" "cached-container-linux-install" {
     var.kernel_args,
   ])
 
-  container_linux_config = data.template_file.cached-container-linux-install-configs[count.index].rendered
+  container_linux_config = data.ct_config.cached-container-linux-install-configs[count.index].rendered
 }
 
-data "template_file" "cached-container-linux-install-configs" {
+data "ct_config" "cached-container-linux-install-configs" {
```
Comment (invidian) [inline assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/profiles.tf:80]: Again, `ct_config` shouldn't be used here.

**Review by invidian — CHANGES_REQUESTED** [review]

File: assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/profiles.tf (hunk 4 above)
Comment (knrt10) [reply assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/profiles.tf:80]: Isn't this os related?

Comment (invidian) [reply assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/profiles.tf:80]: Ah yes, my bad, sorry :pray:

**Review by invidian — DISMISSED** [review]
Looks good

**Review by invidian — APPROVED** [review]

**Review by invidian — APPROVED** [review]

**Review by surajssd — APPROVED** [review]
LGTM

**Pull request merged**